src/main/resources/db/migration/V001.001__datasets_array.sql

```diff
@@ -0,0 +1,36 @@
+-- See the NOTICE file distributed with this work for additional information
+-- regarding copyright ownership.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Updates trial.additional_info to have a datasets array instead of just observationDatasetId.
+-- Leaves observationDatasetId in place out of an abundance of caution.
+DO
+$$
+BEGIN
+    UPDATE
+        trial
+    SET
+        additional_info = additional_info
```

Comment:
If the new code was run prior to running this db migration for some reason, would the trial datasets array be overwritten with the old datasetId value? If so, do we need to do any check in the migration or just leave it up to process?

Reply:
I think I'll update the migration to be safe.

Reply:
Done, [8e06edf](https://github.com/Breeding-Insight/brapi-Java-TestServer/pull/33/commits/8e06edf39a0bcc272678a07350980fe14c757070).

I tested this change by creating an experiment with the old code, then checking out the bi-web and bi-api feature/BI-2109 branches, creating two experiments and a sub-entity dataset, then checking out the BrAPI-Java-TestServer feature/BI-2109 branch and running it to ensure the migration added the `datasets` key for the "legacy" data and preserved the `datasets` key for the "new" data.

Reply:
sounds good